### PR TITLE
Feature/4544 smart boolean output parser

### PR DESCRIPTION
**Pull request by JacobFV**
# Added `SmartBooleanOutputParser`

Implements #4544

## Before submitting

- Will be deomnstrating its use in the concise api notebook #4250 (coming soon)

## Who can review?

@hwchase17 : This PR is one bead on the chain. I'll mesage you on slack when they're all ready.

### CODE REVIEW

File: langchain/output_parsers/smart_boolean.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        Args:
+            text: output of language model
+
+        Returns:
+            boolean
+
+        """
+        for word in text.split():
+            word = word.strip("\r\n\t \"',.:;!?[]{}|/\\").lower()
```
Comment (dev2049) [inline langchain/output_parsers/smart_boolean.py:31]: wonder if using re would be a bit more robust + succinct (replace yes/no with true_val/false_val)

Screenshot 2023-05-11 at 8 35 29 PM

File: tests/unit_tests/output_parsers/test_smart_boolean_parser.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    # Test valid input
+    result = parser.parse("YES")
+    assert result is True
+
+    # Test valid input
+    result = parser.parse("NO")
+    assert result is False
+
+    # Test valid input
```
Comment (hwchase17) [inline tests/unit_tests/output_parsers/test_smart_boolean_parser.py:15]: lets add tests for when multiple yes/no are present

File: tests/unit_tests/output_parsers/test_smart_boolean_parser.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+def test_smart_boolean_output_parser_parse() -> None:
+    parser = SmartBooleanOutputParser()
+
+    # Test valid input
+    result = parser.parse("YES")
+    assert result is True
+
+    # Test valid input
+    result = parser.parse("NO")
```
Comment (hwchase17) [inline tests/unit_tests/output_parsers/test_smart_boolean_parser.py:12]: these should all be separate test cases

**Comment by baskaryan** [comment]
stale, closing for now but feel free to reopen if you return to it @JacobFV

**Pull request closed without merging**